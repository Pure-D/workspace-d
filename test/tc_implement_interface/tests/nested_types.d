module nested_types;

class CoolClassImpl : CoolClass
{

}

abstract class CoolClass
{
	struct Helper
	{
		int x, y;

		void foo()
		{
		}
	}

	class Helper2
	{
		string foo;

		void bar();
	}

	void toImplement();
	Helper accessHelper();
	Helper2 accessHelper2();
}
