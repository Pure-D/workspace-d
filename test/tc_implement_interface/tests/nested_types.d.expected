implement 48
!void foo
!void bar
void toImplement()
CoolClass.Helper accessHelper()
CoolClass.Helper2 accessHelper2()